### PR TITLE
Send quiz functionality

### LINKED ISSUE #225: Spiąć funkcjonalności związane z quizem
Trzeba spiąć ze sobą front i bak w ficzurach:
- wysyłanie quizu przez wykładowcę
- odbieranie quizu przez studenta
- wysyłanie odpowiedzi przez studenta
- odbieranie odpowiedzi przez wykładowcę
- wysyłanie poprawnych odpowiedzi do studenta przez wykładowcę

### LINKED ISSUE #272: Naprawić widok quizu po stronie studenta
nie updatują się odpowiedzi

### PR DESCRIPTION
**Pull request by rozchlastywacz**
@Karolk99 code changes are needed - #272 

sending quiz from lecturer to student works, sending answers from student also works
image
 

closes #225 if #272 fixed

### CODE REVIEW

**Review by Petros9 — APPROVED** [review]

File: src/student/components/quizView/QuestionsList.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import { useContext, useEffect } from "react";
+import { useSocket } from '../../services/SocketService';
 
 export function QuestionsList() {
     let answers = new Map();
-    let quiz = testData();
-
+    const [quiz, setQuiz] = useState(testData());
+    const [quizID, setQuizID] = useState("");
+    const store = useContext(StoreContext);
```
Comment (Derstilon) [inline src/student/components/quizView/QuestionsList.tsx:20]:
```suggestion
```
Store is no longer used in this component

**Review by Derstilon — COMMENTED** [review]
Everything works for me. Let me know when the student side will be finished to approve or if those changes will come later in another PR

**Review by Derstilon — APPROVED** [review]
Conflicts were explosive
image

File: src/lecturer/components/sendQuizView/SendQuizView.tsx [hunk 2]
```diff
@@ -41,26 +41,30 @@ function getSteps() {
...
+    const [time, setTime] = useState<number>(store.sendQuiz.timeSeconds ?? 0);
     const [quiz, setQuiz] = useState<boolean>(Boolean(store.sendQuiz.quiz));
     const [checked, setChecked] = useState<boolean>(time === 0);
     const [randomStudentsNumber, setRandomStudentsNumber] = useState<string>();
     const [timerWait, setTimerWait] = useState<NodeJS.Timeout>();
     const [clock, setClock] = useState(0);
 
 
+
+
```
Comment (ostatni5) [inline src/lecturer/components/sendQuizView/SendQuizView.tsx]: too many empty lines

**Review by ostatni5 — CHANGES_REQUESTED** [review]
The timer is not connected to the received quiz 

obraz
obraz

**Review by ostatni5 — APPROVED** [review]

**Pull request merged**